core/src/main/java/com/datastax/oss/driver/internal/core/loadbalancing/helper/LocalDcHelper.java

```diff
@@ -0,0 +1,35 @@
+/*
+ * Copyright DataStax, Inc.
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ * http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+package com.datastax.oss.driver.internal.core.loadbalancing.helper;
+
+import java.util.Optional;
+
+public interface LocalDcHelper {
```

Comment:
Actually this interface is fundamentally equivalent to a Supplier (or, even more generally, a Function).  Might be worth considering changing the impl of newLocalDcHelper() to return a Supplier and move the impl of the concrete classes into lambdas within the various LBP classes.  The advantage is that the entire logic for a given LBP resides within the LBP class rather than being a combination of LBP class + local DC helper class.

Again, not required by any means... just something to consider.

Reply:
From my experience, it is better to "name" the Supplier to give it a specific type.
So if the constructor is taking LocalDcHelper (that happens to be supplier) it is better to have the strong type to allow clients to pass only `LocalDcHelper` instead of wider type (`Supplier`). It is a similar case to having the Name of a person that is a `String` as well. For strong typing, it is better to create an explicit data class name as `class Name{}` instead of having String API.

Reply:
> Might be worth considering changing the impl of newLocalDcHelper() to return a Supplier and move the impl of the concrete classes into lambdas within the various LBP classes. 

This would incur in some code duplication, e.g. if a grand-child class wants to expose the same behavior as its grand-parent, but the parent has a different behavior. This actually happens e.g. for the DSE LBPs.

Reply:
> From my experience, it is better to "name" the Supplier to give it a specific type.

Agreed, and that also allows for better documentation of method contracts.

Reply:
I'm far from attached to the idea... I probably wouldn't even have mentioned it if there hadn't been an earlier conversation about inheritance.  And in this case it's trickier since there is explicit inheritance in the helper hierarchy as well as the LBP policy directly.

That said, this could be worked around.  The existing impls largely do some additional processing if the base impl couldn't determine a DC; these could be easily implemented via fn declarations which include the originally computed value as an arg.

I understand the point about preferring named types.  But in this case the differences in the subtypes are relatively small... and you're only modifying the behaviour of another function on the interface in any meaningful ways.  It seems like using `Function<Map<UUID, Node>, Optional<String>>` to represent that functionality is a more natural (to say nothing of more obvious) way to represent the interaction going on here than an explicit type hierarchy.

Regardless, what's there now definitely does work so I'll not belabor the point. :)